specviz/plugins/model_editor/model_editor.py

```diff
@@ -101,19 +145,60 @@ def _on_remove_model(self):
             selected_idx = indexes[0]
             self.model_tree_view.model().removeRow(selected_idx.row())
 
-    def _add_fittable_model(self, model):
-        idx = self.model_tree_view.model().add_model(model())
+    def _add_fittable_model(self, model_type):
+        if issubclass(model_type, MODELS['Polynomial1D']):
+            text, ok = QInputDialog.getInt(self, 'Polynomial1D',
+                                           'Enter Polynomial1D degree:')
+            if ok:
+                # Create a new astropy model class definition with the degree
+                # data already embedded
```

Comment:
Why is it necessary to do this? Why can't we just create an instance of Astropy's `Polynomial1D` here?